python/triton/runtime/autotuner.py

```diff
@@ -135,6 +146,11 @@ def kernel_call():
             return float("inf") if self.use_cuda_graph else [float("inf"), float("inf"), float("inf")]
 
     def run(self, *args, **kwargs):
+        for hook in self.pre_run_hooks:
+            result = hook(self, *args, **kwargs)
```

Comment:
If you add a callback here, not all arguments are available yet. Are you sure it's useful to you?

Reply:
What are the arguments that aren't available? If you're referring to the args from the configs, our plan was to extract them from the Autotuner object passed to the hook

Reply:
In that case, isn't it better to avoid using the autotuner? Or you are considering about tracing user code which has autotuner in it?

Reply:
We want to trace user code that has the autotuner in it

Reply:
OK I get it. I'll mark it as resolved for now.